jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -129,90 +90,50 @@ void reportSpan(Span span) {
-            // match first on concrete classes
-            do {
-                if (codecMap.containsKey(c)) {
-                    return codecMap.get(c);
-                }
-                c = c.getSuperclass();
-            } while (c != null);
-            // match second on interfaces
-            for (Class<?> iface : carrierType.getInterfaces()) {
-                if (codecMap.containsKey(iface)) {
-                    return codecMap.get(iface);
-                }
-            }
-            throw new IllegalArgumentException("no registered " + type + " for " + carrierType.getName());
-        }
-
-        // Lookup returns an Object reference because it must return both Injectors, and Extractors.
-        // In theory this cast is unsafe, but in reality the code won't abuse this lack of type safety.
-        @SuppressWarnings("unchecked")
-        <T> Injector<T> getInjector(Class<?> carrierType) {
-            return (Injector<T>) lookup(carrierType, injectors, "injector");
-        }
-
-        // Lookup returns an Object reference because it must return both Injectors, and Extractors.
-        // In theory this cast is unsafe, but in reality the code won't abuse this lack of type safety.
-        @SuppressWarnings("unchecked")
-        <T> ExtractorFactory<T> getExtractor(Class<?> carrierType) {
-            return (ExtractorFactory<T>) lookup(carrierType, extractors, "extractor");
-        }
-
-        public <T> Injector<T> register(Class<T> carrierType, Injector<?> injector) {
-            if (injectors.containsKey(carrierType)) {
-                return null;
-            }
-
-            // We store injectors with multiple carrierTypes, but only return the injector for the carrierType
-            // that we just registered.  Thus this is type unsafe.  However logically we know that the Injector
-            // type return the type we just associated with its key.
-            @SuppressWarnings("unchecked")
-            Injector<T> receivedInjector =  (Injector<T>) injectors.put(carrierType, injector);
-            return receivedInjector;
-        }
-
-        public <T> ExtractorFactory register(Class<T> carrierType, ExtractorFactory extractor) {
-            if (extractors.containsKey(carrierType)) {
-                return null;
-            }
-
-            return extractors.put(carrierType, extractor);
+    public <T> io.opentracing.SpanContext extract(Format<T> format, T carrier) {
+        Extractor<T> extractor = registry.getExtractor(format);
+        if (extractor == null) {
+            throw new UnsupportedFormatException(format);
         }
+        return extractor.extract(carrier);
     }
 
-    private abstract class AbstractSpanBuilder implements io.opentracing.Tracer.SpanBuilder {
+    private class SpanBuilder implements io.opentracing.Tracer.SpanBuilder {
 
         private String operationName = null;
         private long start;
+        private SpanContext firstParent;
```

Comment:
fair enough, I copied it from another project, but will rename to `parent`. We do not support multiple parents at this time anyway.